tests/sls/homu/valid_repos.py

```diff
@@ -0,0 +1,59 @@
+import toml
+import requests
+from tests.util import Failure, Success
+
+
+def repo_pager(endpoint, repoblock, auth):
+    """
+    recursively consumes an endpoint and an empty (self-populating) list
+    it will exit when it fails to find the 'next' key in the links dict
+    """
+    req = requests.get(endpoint, headers=auth)
+    repoblock += req.json()
+    try:
+        next_req_endpoint = req.links['next']['url']
+        repo_pager(next_req_endpoint, repoblock, auth)
+    except KeyError:
+        return repoblock
+
+
+def run():
+    # these try and except blocks are largely to compensate for potential
+    # upstream problems with requests and toml
+
+    try:
+        homu_cfg = toml.load('/home/servo/homu/cfg.toml')
+    except Exception as e:
+        return Failure('Unable to read the homu cfg.toml', str(e))
+    # list of repos in homu build configuration
+    try:
+        configured_repos = homu_cfg['repo']
+        keys = configured_repos.keys()
+        # in the event that libraries outside servo org
+        homu_repos = ["https://github.com/"+configured_repos[i]['owner']+'/'+i
+                      for i in keys]
+    except Exception as e:
+        return Failure('Unable to construct list of homu repos', str(e))
+    try:
+        access_token = homu_cfg['github']['access_token']
+        auth = {'Authorization': 'token '+access_token}
+        req_list = []
+        repo_pager('https://api.github.com/orgs/servo/repos', req_list, auth)
+        # grok out the html_url's
+        # when the github api fails to authenticate, the html_url key will not
+        # exist in the req dicts
+        gh_repos = [req['html_url'] for req in req_list]
+    except TypeError as e:
+        return Failure('Unable to authorize to github API', str(e))
```

Comment:
After https://github.com/servo/saltfs/pull/606/commits/d83bedf34bb3baa3e30104a8c916e384044f89fb I feel that the need to catch this exception is significantly diminished, since the authorization check is now handled in the repo_pager function.